gw_eccentricity/eccDefinition.py

```diff
@@ -654,11 +697,97 @@ def process_data_dict(self,
+            # dict via it. Similary for the times using "t".
+            intertial_zeroecc_data_dict = {"t": data_dict["t_zeroecc"],
+                                           "hlm": data_dict["hlm_zeroecc"]}
+            coprecessing_zeroecc_data_dict = get_coprecessing_data_dict(
+                intertial_zeroecc_data_dict)
+            # update the data_dict by replacing the inertial frame zeroecc
+            # modes with coprecessing frame zeroecc modes.
+            data_dict.update(
+                {"hlm_zeroecc": coprecessing_zeroecc_data_dict["hlm"]})
+
+        # if hlm is not in data_dict, get it from amplm and phaselm.
+        # We need to provide hlm to get the rotated modes.
+        if "hlm" not in data_dict:
+            amplm_dict = self.get_amplm_from_dataDict(data_dict)
+            phaselm_dict = self.get_phaselm_from_dataDict(data_dict)
+            hlm_dict = {}
+            # check if "hlm_zeroecc" is not in data_dict but amplm_zeroecc is
+            # in data_dict
+            if "amplm_zeroecc" in data_dict and "hlm_zeroecc" not in data_dict:
+                add_hlm_zeroecc = True
+                hlm_zeroecc_dict = {}
+            else:
+                add_hlm_zeroecc = False
+            # combine amplm and phaselm to get hlm
+            for k in amplm_dict["amplm"]:
+                hlm_dict.update({k: amplm_dict["amplm"][k]
+                                 * np.exp(-1j * phaselm_dict["phaselm"][k])})
+                if add_hlm_zeroecc:
+                    hlm_zeroecc_dict.update(
+                        {k: amplm_dict["amplm_zeroecc"][k]
+                         * np.exp(-1j * phaselm_dict["phaselm_zeroecc"][k])})
+            inertial_ecc_data_dict = {"t": data_dict["t"], "hlm": hlm_dict}
+            coprecessing_ecc_data_dict = get_coprecessing_data_dict(
+                inertial_ecc_data_dict)
+            data_dict.update(coprecessing_ecc_data_dict)
+            # remove amplm, phaselm because these are in the inertial frame
+            # and are given priority over hlm when using data for egw
+            data_dict.pop("amplm", None)
+            data_dict.pop("phaselm", None)
+            if add_hlm_zeroecc:
+                inertial_zeroecc_data_dict = {
+                    "t": data_dict["t_zeroecc"], "hlm": hlm_zeroecc_dict}
+                coprecessing_zeroecc_data_dict = get_coprecessing_data_dict(
+                    inertial_zeroecc_data_dict)
+                data_dict.update(
+                    {"hlm_zeroecc": coprecessing_zeroecc_data_dict["hlm"]})
+                # remove amplm_zeroecc, phaselm_zeroecc because these are in
+                # the inertial frame and are given priority over hlm_zeroecc
+                # when using data for egw
+                data_dict.pop("amplm_zeroecc", None)
+                data_dict.pop("phaselm_zeroecc", None)
+        # remove inertial omegalm. The coprecessing omegalm should be computed
+        # from the coprecessing phase later.
```

Comment:
Print a warning that you are doing this and why.

Reply:
Added warning.